src/altinn-app-frontend/src/features/form/containers/Form.test.tsx

```diff
@@ -152,6 +152,75 @@ describe('Form', () => {
     expect(screen.getByText('1. FormLayout')).toBeInTheDocument();
   });
 
+  it('should not render ErrorReport when there are no validation errors', () => {
+    renderForm(mockComponents);
+    expect(screen.queryByTestId('ErrorReport')).not.toBeInTheDocument();
+  });
+
+  it('should render ErrorReport when there are validation errors', () => {
+    renderForm(
+      mockComponents,
+      mockValidations({
+        component1: {
+          simpleBinding: {
+            errors: ['some error message'],
+          },
+        },
+      }),
+    );
+    expect(screen.getByTestId('ErrorReport')).toBeInTheDocument();
+  });
+
+  it('should render ErrorReport when there are unmapped validation errors', () => {
+    renderForm(
+      mockComponents,
+      mockValidations({
+        unmapped: {
+          simpleBinding: {
+            errors: ['some error message'],
+          },
+        },
+      }),
+    );
+    expect(screen.getByTestId('ErrorReport')).toBeInTheDocument();
+  });
+
+  it('should separate NavigationButtons and display them inside ErrorReport', () => {
+    renderForm(
+      [
+        ...mockComponents,
+        {
+          id: 'bottomNavButtons',
+          type: 'NavigationButtons',
+        },
+      ],
+      mockValidations({
+        component1: {
+          simpleBinding: {
+            errors: ['some error message'],
+          },
+        },
+      }),
+    );
+    const errorReport = screen.getByTestId('ErrorReport');
+    expect(errorReport).toBeInTheDocument();
+
+    // This also asserts the buttons are only found once
+    const navButtons = screen.getByTestId('NavigationButtons');
+    expect(navButtons).toBeInTheDocument();
+
+    const recurseUp = (el: HTMLElement) => {
+      if (!el.parentElement || el.parentElement === el) {
+        return el;
+      }
+      if (el.parentElement.dataset['testid'] === 'ErrorReport') {
+        return el.parentElement;
+      }
+      return recurseUp(el.parentElement);
+    };
+    expect(recurseUp(navButtons)).toEqual(errorReport);
```

Comment:
Hmm, This code isn't just testing that `NavigationButtons` exist, but that it is a child of the `ErrorReport` component. If you look at `extractBottomButtons` in `Form.tsx`, it will remove the `NavigationButtons` button from the layout and render it as a child of `ErrorReport` when there are errors in the form.

Reply:
Ah, I got fooled from the comment and didnt read thoroughly enough - comments are evil 😂

Anyway, if you want to ensure the buttons are rendered inside something else, you could use [`within`](https://testing-library.com/docs/dom-testing-library/api-within/) for this instead of the custom code. IMO that would be preferred, as it would be a bit easier to understand what is going on, and its an established pattern when using RTL 🙂

Something like this should work I think: 
```jsx
expect(within(errorReport).getByTestId('NavigationButtons')).toBeInTheDocument();
```

Reply:
Nice! I believe I googled for this quickly, but didn't find anything. I'll rewrite the test to use this instead. Comments aren't evil, but they can't replace reading the code.. :wink: